generation/check-for-git-changes.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e
+
+if [[ $(git diff) ]]; then
+    echo "Git shows that there are changes."
```

Comment:
Please be more explicit in these logs. 
1. What happened ?
2. Why did it happen ?
3. What should the user do now in order to resolve it ?